### PR TITLE
[#61][#18] Add `Hexadecimal` and `HexColor` modules

### LINKED ISSUE #18: Add Hexadecimal string module
See: https://github.com/validatorjs/validator.js/blob/master/src/lib/isHexadecimal.js

### LINKED ISSUE #61: Add HexColor module
See: https://github.com/validatorjs/validator.js/blob/master/src/lib/isHexColor.js

### PR DESCRIPTION
**Pull request by newswim**
This PR does the following:

- Adds `Hexadecimal` and `HexColor` modules
- Bumps package version to `0.0.3`
- Adds a check for Node v18 to the build process
- Fixes some small typos in some of the string tests

Closes #61 
Closes #18

### CODE REVIEW

File: src/string/HexColor.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+ */
+export type SchemableParams2<S extends URIS2> = Kind2<S, string, HexColor>
+
+/**
+ * @since 0.0.3
+ * @category Model
+ */
+export type SchemableParams2C<S extends URIS2> = Kind2<S, unknown, HexColor>
+
+const hexcolor = /^#?([0-9A-F]{3}|[0-9A-F]{4}|[0-9A-F]{6}|[0-9A-F]{8})$/i
```
Comment (jacob-alford) [inline src/string/HexColor.ts:62]: 💬 you might add an `internal` comment here, but I'm not going to block on it

File: tests/HexColor.test.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+      expect(result._tag).toBe(expectedTag)
+    })
+  })
+
+  describe('Encoder', () => {
+    test.each(validStrings)(
+      'encoding a decoded value yields original value',
+      original => {
+        const roundtrip = pipe(
```
Comment (jacob-alford) [inline tests/HexColor.test.ts:29]: 👍  Very nice!  It's going to be super nice once we have Arbitraries set up too!

**Review by jacob-alford — APPROVED** [review]
🥇 great work!  Thanks for bumping the version

**Review by jacob-alford — APPROVED** [review]
LGTM 🎉

File: src/string/HexColor.ts (hunk 1 above)
Comment (newswim) [reply src/string/HexColor.ts:66]: Oops! Added.

**Pull request merged**